Assets/src/WeaponReload.cs

```diff
@@ -8,51 +8,69 @@ public class WeaponReload : MonoBehaviour {
   public SoundController SoundController;
   public HUDBar ReloadBar;
   public MovementManager MovementManager;
-
-  IntervalTimer ReloadTimer;
+  private IntervalTimer ReloadTimer;
+  private const float movementSpeedReduction = 0.5f;
 
 	// Use this for initialization
 	void Start () {
-    ReloadTimer = new IntervalTimer() {
-      TimeClass = "PlayingTime"
-    };
-
-    // Reduce speed by half when reload is InProgress
+    InitializeReloadTimer();
+	}
+	
+  private void InitializeReloadTimer() {
+    ReloadTimer = new IntervalTimer(){TimeClass = "PlayingTime"};
     MovementManager.SpeedModifiers.Add(
-      s => InProgress() ? 0.5f * s : s
+      speed => InProgress() ? movementSpeedReduction * speed : speed
     );
+  }
 
-	}
-	
-	// Update is called once per frame
 	void Update () {
     if (!StateManager.Playing)
       return;
-
-    ReloadTimer.IfElapsed( CurrentWeapon().Reload );
-
+    HandleReloading();
+    
     if ( WrappedInput.GetButtonDown("Reload") && !InProgress() && CanReload() )
       BeginReload();
 
-    if ( InProgress() ) {
-      if ( CanReload() ) {
-        ReloadBar.Progress = ReloadTimer.Progress();
-      } else {
-        ReloadTimer.Stop();
-        SoundController.Stop();
-      }
-    }
-
+    UpdateReloadProgress();
     ReloadBar.Visible = InProgress();
 	}
 
+  private void UpdateReloadProgress() {
+    if (!InProgress())
+      return;
+
+    if (CanReload())
+      ReloadBar.Progress = ReloadTimer.Progress();
+    else
+      StopReloading();
+  }
+
+  private void HandleReloading() {
+    ReloadTimer.IfElapsed( CurrentWeapon().Reload );
+
+    if (UserTriggeredReload() && CanReload())
+      BeginReload();
+
+    UpdateReloadProgress();
+  }
+  private bool UserTriggeredReload() {
+    return WrappedInput.GetButtonDown("Reload") && !InProgress();
+  }

```

Comment:
There seems to be some unintended duplication here? It's a little hard to make sense of what's going on with these changes.